desktop-exporter/otlp_payload_processor.go

```diff
@@ -45,14 +73,34 @@ func extractSpans(_ context.Context, traces ptrace.Traces) []SpanData {
 
 			for si := 0; si < scopeSpan.Spans().Len(); si++ {
 				span := scopeSpan.Spans().At(si)
-				spanData := aggregateSpanData(span, scopeData, resourceData)
+				eventData := extractEvents(span.Events())
+				linkData := extractLinks(span.Links())
+				spanData := aggregateSpanData(span, eventData, linkData, scopeData, resourceData)
 				extractedSpans = append(extractedSpans, spanData)
 			}
 		}
 	}
 	return extractedSpans
 }
 
+func extractEvents(events ptrace.SpanEventSlice) []EventData {
+	eventDataSlice := make([]EventData, 0, events.Len())
+	for eventIndex := 0; eventIndex < events.Len(); eventIndex++ {
+		eventDataSlice = append(eventDataSlice, aggregateEventData(events.At(eventIndex)))
+	}
+
+	return eventDataSlice
+}
+
+func extractLinks(links ptrace.SpanLinkSlice) []LinkData {
+	linkDataSlice := make([]LinkData, 0, links.Len())
+	for linkIndex := 0; linkIndex < links.Len(); linkIndex++ {
+		linkDataSlice = append(linkDataSlice, aggregateLinkData(links.At(linkIndex)))
+	}
+
+	return linkDataSlice
```

Comment:
Nope, I can't think of a cleaner way to write this. This is the way

![image](https://user-images.githubusercontent.com/175496/198145349-eb00f85b-9c49-4dd9-87ab-74de5fcc4fe4.png)